pil-analyzer/src/type_inference.rs

```diff
@@ -168,13 +229,13 @@ impl TypeChecker {
 
         // Now we check for all symbols that are not declared as a type scheme that they
         // can resolve to a concrete type.
-        for (name, (source_ref, declared_type)) in &self.declared_types {
```

Comment:
Instead of accessing `self.declared_types` directly, return an iterator over all concrete symbols.